src/ol/source/ogcTileUtil.js

```diff
@@ -184,6 +188,42 @@ export function getVectorTileUrlTemplate(
     }
   }
 
+  if (collections && collections.length) {
+    // According to conformance class
+    // http://www.opengis.net/spec/ogcapi-tiles-1/1.0/conf/collections-selection
+    // commata in the identifiers of the `collections` query parameter
+    // need to be URLEncoded, while the commata separating the identifiers
+    // should not.
+    let url;
+    let isAbsolute = true;
+    try {
+      url = new URL(tileUrlTemplate);
+    } catch (e) {
+      isAbsolute = false;
+      url = new URL(tileUrlTemplate, 'https://example.com');
+    }
```

Comment:
Instead of using try/catch and differentiating between relative and absolute URLs, you could simplify this to
```suggestion
    const url = new URL(tileUrlTemplate, window.location.href);
```